server/lib/flow-event.js

```diff
@@ -88,6 +94,21 @@ module.exports = (req, metrics, requestReceivedTime) => {
 
     logFlowEvent(event, metrics, req);
   });
+
+  const navigationTiming = metrics.navigationTiming;
+  if (emitPerformanceEvents) {
+    PERFORMANCE_PROPERTIES.forEach(key => {
+      const timing = navigationTiming[key];
+      if (timing > 0) {
+        const offset = timing - navigationTiming.navigationStart;
```

Comment:
> One thing to watch out for, I believe these values can be `null` or `undefined`.

That's (one of the reasons) why I test `timing > 0` above. It's fine that we won't have this data for all flows, I see it more as an additional, secondary datapoint for us to consult when trying to understand certain trends.

> I think all these values are already offsets

I copied the subtraction of `navigationStart` [from MDN](https://developer.mozilla.org/en-US/docs/Web/API/Navigation_timing_API#Examples). Although they're certainly offsets in current Firefox, I don't know if that's always been/will be true across all user agents. I figured the subtraction was there to provide a measure of robustness against implementation differences (including historical).